### PR TITLE
Rationalize how defaults are used when constructing Mixer and tracing cobra args

### PR DESCRIPTION
**Pull request by geeknoid**
This uses the same pattern as pkg/log and other places to avoid specifying default
values in two places.

### CODE REVIEW

File: mixer/pkg/server/args.go [hunk 1]
```diff
@@ -121,6 +121,7 @@ func NewArgs() *Args {
 		TracingOptions:                tracing.NewOptions(),
 		LivenessProbeOptions:          &probe.Options{},
 		ReadinessProbeOptions:         &probe.Options{},
+		EnableProfiling:               true,
```
Comment (mandarjog) [inline mixer/pkg/server/args.go]: we should rename `NewArgs()` to `DefaultArgs()`

**Comment by geeknoid** [comment]
OK, PTAL.

**Pull request merged**